docs/ibc/middleware/integration.md

```diff
@@ -19,23 +19,22 @@ The order of middleware **matters**, function calls from IBC to the application
 
 // middleware 1 and middleware 3 are stateful middleware, 
 // perhaps implementing separate sdk.Msg and Handlers
-mw1Keeper := mw1.NewKeeper(storeKey1)
-mw3Keeper := mw3.NewKeeper(storeKey3)
+mw1Keeper := mw1.NewKeeper(storeKey1, ..., ics4Wrapper: channelKeeper, ...) // in stack 1 & 3
+// middleware 2 is stateless
+mw3Keeper1 := mw3.NewKeeper(storeKey3,..., ics4Wrapper: mw1Keeper, ...) //  in stack 1
+mw3Keeper2 := mw3.NewKeeper(storeKey3,..., ics4Wrapper: channelKeeper, ...) //  in stack 2
 
 // Only create App Module **once** and register in app module
 // if the module maintains independent state and/or processes sdk.Msgs
 app.moduleManager = module.NewManager(
   ...
   mw1.NewAppModule(mw1Keeper),
-  mw3.NewAppModule(mw3Keeper),
+  mw3.NewAppModule(mw3Keeper1),
+  mw3.NewAppModule(mw3Keeper2),
   transfer.NewAppModule(transferKeeper),
   custom.NewAppModule(customKeeper)
 )
 
-mw1IBCModule := mw1.NewIBCModule(mw1Keeper)
-mw2IBCModule := mw2.NewIBCModule() // middleware2 is stateless middleware
-mw3IBCModule := mw3.NewIBCModule(mw3Keeper)
-
 scopedKeeperTransfer := capabilityKeeper.NewScopedKeeper("transfer")
 scopedKeeperCustom1 := capabilityKeeper.NewScopedKeeper("custom1")
 scopedKeeperCustom2 := capabilityKeeper.NewScopedKeeper("custom2")
```

Comment:
These scoped keepers are created here, but not used anywhere else... Should we show pseudocode as well for the keepers that use them?

Reply:
I had not paid too much attention here to the scoped keepers but I think you're correct

Reply:
Updated this